### ISSUE TITLE
lightsource shade bug

### ISSUE THREAD

**Issue by leomiquelutti** [jobar8/graphics#4]
Hi @jobar8 

I am trying to use your (fantastic) ```graphics.py```, but I've been facing a strange behavior that I cannot explain or, worse, solve. 

I tried ```imshow_hs``` with 3 different data sets, but there is one that always crashes the code when ```hs=True```, i.e. shading is on.

At some point, I believed that it was a ```matplotlib``` issue, so I started a _Jupyter Notebook_ to do a bug report. Turns out that I don't know now the origin of the problem, so I thought to ask your help. 

Below is the code that works with your dataset (mag4017) and then crashes with mine. Same input parameters.
Dataset that works  &rarr; [mag4017](https://1drv.ms/u/s!AvWFfSwwsz6gloRXDZeHRtpQWJrysw?e=JRTTZK)
Dataset that *does not work* &rarr; [my_data](https://1drv.ms/u/s!AvWFfSwwsz6gloRYkglNSu9F1V-klg?e=48MouQ)

```
import numpy as np
import matplotlib.pyplot as plt
import graphics

data_good = np.load('mag4017.npy')
fig,ax = plt.subplots(figsize=(12,6))
graphics.imshow_hs(data_good,ax,cmap=plt.cm.gist_earth,cmap_norm='equalize',hs=True,zf=2,levels=32,colorbar=False,
                   cb_ticks='stats',nSigma=2,azdeg=45,altdeg=45,blend_mode='overlay',alpha=0.7,
                   extent=extent, origin='lower')

data_bad = np.load('my_data.npy')
fig,ax = plt.subplots(figsize=(12,6))
graphics.imshow_hs(data_bad,ax,cmap=plt.cm.gist_earth,cmap_norm='equalize',hs=True,zf=2,levels=32,colorbar=False,
                   cb_ticks='stats',nSigma=2,azdeg=45,altdeg=45,blend_mode='overlay',alpha=0.7,
                   extent=extent, origin='lower')
```
It doesn't matter if you choose ```blend_mode=alpha``` or whatever.

In the file attached, there is a _Jupyter Notebook_ that also shows a function that I did to try to "mimic" the problem, but it runs well where ```graphics.py``` crashes.

To sum up, I am waaayyyyy lost!

[shade bug report.zip](https://github.com/jobar8/graphics/files/4908645/shade.bug.report.zip)

**Comment by leomiquelutti** [comment]
Solved with a new environment installation.

**Comment by jobar8** [comment]
Hi Leonardo,
Sorry I did not reply to your message. I haven't got much time to look at this project these days.
I am glad you found a solution!
Joseph
